bluesky_queueserver/server/server.py

```diff
@@ -102,140 +121,284 @@ class REPauseOptions(str, Enum):
+    May be called to get some response from the server. Currently returns status of RE Manager.
     """
     msg = await re_server.send_command(command="")
     return msg
 
 
-@app.get("/get_queue")
-async def get_queue_handler():
+@app.get("/status")
+async def status_handler():
+    """
+    Returns status of RE Manager.
+    """
+    msg = await re_server.send_command(command="status")
+    return msg
+
+
+@app.get("/queue/get")
+async def queue_get_handler():
     """
     Returns the contents of the current queue.
     """
-    msg = await re_server.send_command(command="get_queue")
+    msg = await re_server.send_command(command="queue_get")
     return msg
 
 
-@app.get("/list_allowed_plans_and_devices")
-async def list_allowed_plans_and_devices_handler():
+@app.post("/queue/clear")
+async def queue_clear_handler():
     """
-    Returns the lists of allowed plans and devices.
+    Clear the plan queue.
+    """
+    msg = await re_server.send_command(command="queue_clear")
+    return msg
+
+
+@app.post("/queue/start")
+async def queue_start_handler():
+    """
+    Start execution of the loaded queue. Additional runs can be added to the queue while
+    it is executed. If the queue is empty, then nothing will happen.
+    """
+    msg = await re_server.send_command(command="queue_start")
+    return msg
+
+
+@app.post("/queue/stop/activate")
+async def queue_stop_activate():
+    """
+    Activate the sequence of stopping the queue. Currently running plan will be completed,
+    but the next plan will not be started.
+    """
+    msg = await re_server.send_command(command="queue_stop_activate")
+    return msg
+
+
+@app.post("/queue/stop/deactivate")
+async def queue_stop_deactivate():
     """
-    msg = await re_server.send_command(command="list_allowed_plans_and_devices")
+    Deactivate the sequence of stopping the queue.
+    """
+    msg = await re_server.send_command(command="queue_stop_deactivate")
     return msg
 
 
-@app.post("/add_to_queue")
-async def add_to_queue_handler(payload: dict):
+@app.post("/queue/plan/add")
+async def queue_plan_add_handler(payload: dict):
     """
-    Adds new plan to the end of the queue
+    Adds new plan to the queue
     """
     # TODO: validate inputs!
-    msg = await re_server.send_command(command="add_to_queue", params=payload)
+    msg = await re_server.send_command(command="queue_plan_add", params=payload)
     return msg
 
 
-@app.post("/clear_queue")
-async def clear_queue_handler():
+@app.post("/queue/plan/remove")
+async def queue_plan_remove_handler(payload: dict):
     """
-    Clear the plan queue.
+    Remove plan from the queue
     """
-    msg = await re_server.send_command(command="clear_queue")
+    msg = await re_server.send_command(command="queue_plan_remove", params=payload)
     return msg
 
 
-@app.post("/pop_from_queue")
-async def pop_from_queue_handler():
+@app.get("/queue/plan/get")
```

Comment:
It is odd for a GET to have a body/payload, quoting 

> Server semantics for GET, however, are restricted such that a body, if any, has no semantic meaning to the request.

POSTs without bodies feels odd, but GET with a body is even odder, I would expect this to be a POST, or to simply encode in the URL, the [HTTP 1.1 spec says](https://tools.ietf.org/html/rfc2616#section-9.3):

> The GET method means retrieve whatever information ([...]) is identified by the Request-URI.

Reply:
Can plans not just be retrieved using simple index or identifiers? So `/queue/plan/get/2` for example would get the second plan.

Reply:
I can change it to POST (as remove). My idea was to keep the call to `/queue/plan/get` identical to the call for `/queue/plan/remove' method. Remove method calls 'queue_plan_get' underneath anyway, because items in the queue are not naturally accessed by index: you can pop a given plan, not plan at a given location. I was also thinking of extending the structure of parameters.

Reply:
I think making it POST is far less surprising than a GET that requires a body. I have used many APIs where POST was used on something that doesn't change data on the server but needs to support more complex query parameters than can reasonably be encoded in the URL.